tests/noise/test_conditionals.py

```diff
@@ -0,0 +1,275 @@
+        """Test for BooleanFn builds correct objects"""
+
+        func = qml.BooleanFn(lambda x: x < 10, "less_than_ten")
+
+        assert isinstance(func, qml.BooleanFn)
+        assert func(2) and not func(42)
+        assert repr(func) == "BooleanFn(less_than_ten)"
+
+    def test_noise_conditional_def(self):
+        """Test for BooleanFn builds correct objects"""
+
+        def greater_than_five(x):
+            return x > 5
+
+        func = qml.BooleanFn(greater_than_five)
+
+        assert isinstance(func, qml.BooleanFn)
+        assert not func(3) and func(7)
+        assert repr(func) == "BooleanFn(greater_than_five)"
+
+    def test_and_conditionals(self):
+        """Test for BooleanFn supports bitwise AND"""
+
+        @qml.BooleanFn
+        def is_int(x):
+            return isinstance(x, int)
+
+        @qml.BooleanFn
+        def has_bit_length_3(x):
+            return x.bit_length() == 3
+
+        func = is_int & has_bit_length_3
+
+        assert isinstance(func, And)
+        assert func(4) and not func(2.3)
+        assert repr(func) == "And(is_int, has_bit_length_3)"
+        assert str(func) == "is_int & has_bit_length_3"
+        assert func.bitwise
+
+    def test_or_conditionals(self):
+        """Test for BooleanFn supports bitwise OR"""
+
+        @qml.BooleanFn
+        def is_int(x):
+            return isinstance(x, int)
+
+        @qml.BooleanFn
+        def less_than_five(x):
+            return x < 5
+
+        func = is_int | less_than_five
+
+        assert isinstance(func, Or)
+        assert func(4) and not func(7.5)
+        assert repr(func) == "Or(is_int, less_than_five)"
+        assert str(func) == "is_int | less_than_five"
+        assert func.bitwise
+
+    def test_xor_conditionals(self):
+        """Test for BooleanFn supports bitwise XOR"""
+
+        @qml.BooleanFn
+        def is_int(x):
+            return isinstance(x, int)
+
+        @qml.BooleanFn
+        def has_bit_length_3(x):
+            return x.bit_length() == 3
+
+        func = is_int ^ has_bit_length_3
+
+        assert isinstance(func, Xor)
+        assert not func(4) and func(11)
+        assert repr(func) == "Xor(is_int, has_bit_length_3)"
+        assert str(func) == "is_int ^ has_bit_length_3"
+        assert func.bitwise
+
+    def test_not_conditionals(self):
+        """Test for BooleanFn supports bitwise NOT"""
+
+        def is_int(x):
+            return isinstance(x, int)
+
+        func = ~qml.BooleanFn(is_int)
+
+        assert isinstance(func, Not)
+        assert not func(4) and func(7.5)
+        assert repr(func) == "Not(is_int)"
+        assert str(func) == "~is_int"
+        assert func.bitwise
+
+
+class TestNoiseFunctions:
+    """Test for the Conditional methods"""
+
+    @pytest.mark.parametrize(
+        ("obj", "wires", "result"),
+        [
+            (0, 0, True),
+            ([1, 2, 3], 0, False),
+            (qml.wires.Wires(["street", "fighter"]), "fighter", True),
```

Comment:
😆 are we okey with so random names in PennyLane?

Reply:
tweaked them.